src/providers/data_provider_be.c

```diff
@@ -306,28 +350,23 @@ static void be_check_online_done(struct tevent_req *req)
     }
 }
 
-static void check_if_online(struct be_ctx *be_ctx)
+static void check_if_online(struct be_ctx *be_ctx, int delay)
 {
-    errno_t ret;
-
-    be_run_unconditional_online_cb(be_ctx);
-
-    if (!be_is_offline(be_ctx)) {
-        DEBUG(SSSDBG_TRACE_INTERNAL,
-              "Backend is already online, nothing to do.\n");
-        return;
-    }
+    struct tevent_timer *time_event;
+    struct timeval schedule;
 
     /* Make sure nobody tries to go online while we are checking */
     be_ctx->offstat.went_offline = time(NULL);
 
-    DEBUG(SSSDBG_TRACE_INTERNAL, "Trying to go back online!\n");
-
     be_ctx->check_online_ref_count++;
 
     if (be_ctx->check_online_ref_count != 1) {
         DEBUG(SSSDBG_TRACE_INTERNAL,
               "There is an online check already running.\n");
+        /* Do not have more than ONLINE_CB_RETRY retries in the queue */
+        if (be_ctx->check_online_ref_count > ONLINE_CB_RETRY) {
+            be_ctx->check_online_ref_count--;
+        }
```

Comment:
From my point of view, it would be a little bit more readable:
```
if (be_ctx->check_online_ref_count >= ONLINE_CB_RETRY) {
    return;
}
be_ctx->check_online_ref_count++;
```
but matter of taste of course.
I don't know if debug message is needed at all, but if it is then it could be more informative (including current value of check_online_ref_count).